### PR TITLE
feat(mandatory_parameters): Renamed mandatory to required to be consi…

### PR DESCRIPTION
**Pull request by henrikstranneheim**
…stent with CLI and validate parameters

### This PR fixes:

- See above

### How to test:

- Automatic and continuous test pass

### Expected outcome:
- Installation, unit and integration tests pass

### Review:
- [x] Code review
- [x] New code is executed and covered by tests
- [x] Tests pass

### CODE REVIEW

File: lib/MIP/Definition.pm [hunk 1]
```diff
@@ -309,10 +309,10 @@ sub get_parameter_from_definition_files {
             %parameter,
             check_definition_file(
                 {
-                    define_parameters_path         => $definition_file,
-                    mandatory_definition_file_path => $mandatory_definition_file_path,
-                    not_mandatory_definition_file_path =>
-                      $not_mandatory_definition_file_path,
+                    define_parameters_path        => $definition_file,
+                    required_definition_file_path => $required_definition_file_path,
+                    not_required_definition_file_path =>
```
Comment (jemten) [inline lib/MIP/Definition.pm]: order AB

File: lib/MIP/Parameter.pm [hunk 2]
```diff
@@ -41,17 +41,17 @@ sub check_parameter_hash {
...
 ## Function : Evaluate parameters in parameters hash
 ## Returns  :
-## Arguments: $file_path          => Path to yaml file
-##          : $mandatory_href     => Hash with mandatory key {REF}
-##          : $not_mandatory_href => Hash with non mandatory key {REF}
-##          : $parameter_href     => Hash with parameters from yaml file {REF}
+## Arguments: $file_path         => Path to yaml file
+##          : $required_href     => Hash with required key {REF}
+##          : $not_required_href => Hash with non required key {REF}
+##          : $parameter_href    => Hash with parameters from yaml file {REF}
```
Comment (jemten) [inline lib/MIP/Parameter.pm]: order AB

File: lib/MIP/Parameter.pm [hunk 3]
```diff
@@ -41,17 +41,17 @@ sub check_parameter_hash {
...
 
     my ($arg_href) = @_;
 
     ##Flatten argument(s)
     my $file_path;
-    my $mandatory_href;
-    my $not_mandatory_href;
+    my $required_href;
+    my $not_required_href;
     my $parameter_href;
```
Comment (jemten) [inline lib/MIP/Parameter.pm]: order AB and rest of sub

File: t/check_definition_file.t [hunk 4]
```diff
@@ -64,19 +64,19 @@ my $level = q{mip};
...
+  catfile( dirname($Bin), qw{ definitions required_parameters.yaml } );
 
 my %parameter = check_definition_file(
     {
-        define_parameters_path             => $definition_file_path,
-        mandatory_definition_file_path     => $mandatory_definition_file_path,
-        not_mandatory_definition_file_path => $not_mandatory_definition_file_path,
+        define_parameters_path            => $definition_file_path,
+        required_definition_file_path     => $required_definition_file_path,
+        not_required_definition_file_path => $not_required_definition_file_path,
```
Comment (jemten) [inline t/check_definition_file.t]: order AB

File: t/check_parameter_hash.t [hunk 5]
```diff
@@ -66,30 +66,29 @@ my $definitions_file = catfile( $Bin, qw{ data test_data define_parameters.yaml
...
 my $is_ok = check_parameter_hash(
     {
-        parameter_href     => \%parameter,
-        mandatory_href     => \%mandatory,
-        not_mandatory_href => \%not_mandatory,
-        file_path          => $definitions_file,
+        parameter_href    => \%parameter,
+        required_href     => \%required,
+        not_required_href => \%not_required,
+        file_path         => $definitions_file,
```
Comment (jemten) [inline t/check_parameter_hash.t]: order AB in sub

File: lib/MIP/Definition.pm (hunk 1 above)
Comment (henrikstranneheim) [reply lib/MIP/Definition.pm]: Done

File: lib/MIP/Parameter.pm (hunk 2 above)
Comment (henrikstranneheim) [reply lib/MIP/Parameter.pm]: Done

File: t/check_definition_file.t (hunk 4 above)
Comment (henrikstranneheim) [reply t/check_definition_file.t]: Done

File: t/check_parameter_hash.t (hunk 5 above)
Comment (henrikstranneheim) [reply t/check_parameter_hash.t]: Done

File: lib/MIP/Parameter.pm (hunk 3 above)
Comment (henrikstranneheim) [reply lib/MIP/Parameter.pm]: Done

**Comment by henrikstranneheim** [comment]
Yeah, they do leave a faint trail; 😄

**Pull request merged**